Model/SourcesRepository.php

```diff
@@ -0,0 +1,178 @@
+        SourceSelectionItemInterfaceFactory $sourceSelectionItemInterface
+    ) {
+        $this->sourcesFactory = $sourcesFactory;
+        $this->sourcesResourceModel = $sourcesResourceModel;
+        $this->serializer = $serializer;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->sourceSelectionResultFactory = $sourceSelectionResultFactory;
+        $this->sourceSelectionItemInterface = $sourceSelectionItemInterface;
+
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceModel
+     * @throws NoSuchEntityException
+     */
+    public function getByOrderId(string $orderId): SourceModel
+    {
+        /** @var SourceModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+
+        $this->sourcesResourceModel->load(
+            $sourcesModel,
+            $orderId,
+            'order_id'
+        );
+
+        return $sourcesModel;
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceSelectionResultInterface
+     */
+    public function getSourceSelectionResultByOrderId(string $orderId): SourceSelectionResultInterface
+    {
+        /** @var SourceModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+
+        $this->sourcesResourceModel->load(
+            $sourcesModel,
+            $orderId,
+            'order_id'
+        );
+
+        $sourcesArray = $this->serializer->unserialize($sourcesModel->getSources());
+        $sourceSelectionItems = [];
+
+        foreach ($sourcesArray as $item)
+        {
+            $sourceSelectionItem = $this->sourceSelectionItemInterface->create(
+                [
+                    'sourceCode' => $item['source_code'],
+                    'sku' => $item['SKU'],
+                    'qtyToDeduct' => $item['qty_to_deduct'],
+                    'qtyAvailable' => $item['qty_available']
+                ]
+            );
+
+        $sourceSelectionItems[] = $sourceSelectionItem;
+        }
+
+        $sourceSelectionResult = $this->sourceSelectionResultFactory->create(
+            [
+                'sourceItemSelections' => $sourceSelectionItems,
+                'isShippable' => true
+            ]
+        );
+
+        return $sourceSelectionResult;
+    }
+
+    /**
+     * @param array $sourcesItems
+     * @param string $orderId
+     *
+     * @return SourceModel
+     * @throws CouldNotSaveException
+     */
+    public function save(array $sourcesItems, string $orderId): SourceModel
+    {
+        $sources = [];
+        foreach ($sourcesItems as $item) {
+            $sources[] = [
+                'source_code' => $item->getSourceCode(),
+                'SKU' => $item->getSku(),
+                'qty_to_deduct' => $item->getQtyToDeduct(),
+                'qty_available' => $item->getQtyAvailable()
+            ];
+        }
+
+        $model = $this->getByOrderId($orderId);
+        if (!$model->getId()) {
```

Comment:
I don't think you need this conditional in this instance. Setting the `order_id` the same again would not cause any issue IMO, so:
```php
$model = $this->getByOrderId($orderId);
$model->addData([
    'order_id' => 'order123',
    ...
]);
```
Or even better, once you've added an interface for your model you would be able to `$model->setOrderId` etc